plugins/opa/cmd/root.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright © 2022 FairwindsOps Inc
+
+*/
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	opa "github.com/fairwindsops/insights-plugins/opa/pkg/opa"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "opa",
+	Short: "Fairwinds Insights OPA plugin",
+	Long: `The OPA plugin runs custom OPA policies against existing Kubernetes resources, to create Action Items.
+	See the Fairwinds Insights documentation, at https://insights.docs.fairwinds.com/reports/opa/`,
+	Run: func(cmd *cobra.Command, args []string) {
+		startOPAReporter()
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.opa.yaml)")
+	rootCmd.PersistentFlags().BoolP("debug", "D", false, "Enable debug logging.")
+	viper.BindPFlag("debug", rootCmd.PersistentFlags().Lookup("debug"))
+	rootCmd.PersistentFlags().StringArrayP("target-resource", "r", []string{}, "A Kubernetes target specified as APIGroup[,APIGroup...]/Resource[,Resource...]. For example: apps/Deployments,Daemonsets")
+	viper.BindPFlag("target-resource", rootCmd.PersistentFlags().Lookup("target-resource"))
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := os.UserHomeDir()
+		cobra.CheckErr(err)
+
+		// Search config in home directory with name ".opa" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigType("yaml")
+		viper.SetConfigName(".opa")
```

Comment:
what does this do?

Reply:
It's Viper boilerplate to also use a config-file as optional input to specify command-line options. While I"m not processing it currently (have removed the option from showing up in `--help`), I left the code in here as an option, and because it was generated by Cobra.